### PR TITLE
Need to update HorizontalPodAutoscaler yaml file under section Autoscaling for Memory Utilization

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1801532

### CODE REVIEW

**Comment by mburke5678** [comment]
@weinliu Please take a look.

**Comment by weinliu** [comment]
> @weinliu Please take a look.

I'm going thru the doc and PR step by step. It seems we are not providing exmaples for DeploymentConfig/frontend  and DeploymentConfig/hello-hpa-cpu.
Can we add exmples for them?  Without the RC, hpa can not be created
`Conditions:
  Type         Status  Reason          Message
  ----         ------  ------          -------
  AbleToScale  False   FailedGetScale  the HPA controller was unable to get the target's current scale: deploymentconfigs.apps.openshift.io "hello-hpa-memory" not found
Events:
  Type     Reason          Age                From                       Message
  ----     ------          ----               ----                       -------
  Warning  FailedGetScale  11s (x19 over 9m)  horizontal-pod-autoscaler  deploymentconfigs.apps.openshift.io "hello-hpa-memory" not found
`

**Comment by mburke5678** [comment]
@weinliu I changed the wording to tell users to specify an existing DC or RC, which should solve the problem you hit. This way, we can avoid adding code examples that are not directly related to creating an HPA.
Please let me know if this is sufficient.

**Comment by mburke5678** [comment]
@joelsmith Can you please do a sanity check? The BZ pointed out that one of the HPA examples pointed to a ReplicationController using the 

In the course of fixing that up, I noticed a few other things that I cleaned up. 
Such as, we tell users that they can use the `oc autoscale` command with the `autoscaling/v2beta1` API to scale for memory utilization, which I believe you cannot do. Does the following section need to be removed: 
https://github.com/openshift/openshift-docs/pull/19761/files#diff-86034f495da5bc5b5510f61ae8a74869R139-R168

**Comment by weinliu** [comment]
> @weinliu I changed the wording to tell users to specify an existing DC or RC, which should solve the problem you hit. This way, we can avoid adding code examples that are not directly related to creating an HPA.
> Please let me know if this is sufficient.

@mburke5678 Current updates LGTM. Thanks.

**Comment by joelsmith** [comment]
Looks good. Thanks!

**Comment by mburke5678** [comment]
@weinliu @joelsmith Thanks for the quick review!

File: dev_guide/pod_autoscaling.adoc [hunk 1]
```diff
@@ -94,25 +93,23 @@ xref:../dev_guide/application_health.adoc#dev-guide-application-health[readiness
...
 
 ----
-$ oc autoscale dc/frontend --min 1 --max 10 --cpu-percent=80
+$ oc autoscale dc/frontend --max 10 --cpu-percent=80
 deploymentconfig "frontend" autoscaled
 ----
 
-The above example creates a horizontal pod autoscaler with the following
-definition when using the `autoscaling/v1` version of the horizontal pod autoscaler:
+The example command creates a horizontal pod autoscaler for an exiting DeploymentConfig with the following
```
Comment (bergerhoffer) [inline dev_guide/pod_autoscaling.adoc]: Should exiting be existing?

File: dev_guide/pod_autoscaling.adoc [hunk 2]
```diff
@@ -129,14 +126,13 @@ spec:
...
-<2> The kind of object to scale
-<3> The name of the object to scale
-<4> The API version of the object to scale
-<5> The minimum number of replicas to which to scale down
-<6> The maximum number of replicas to which to scale up
-<7> The percentage of the requested CPU that each pod should ideally be using
-====
+<1> The name of this horizontal pod autoscaler object.
+<2> The kind of object to scale.
+<3> The name of an exisiting object you want to scale.
```
Comment (bergerhoffer) [inline dev_guide/pod_autoscaling.adoc]: exisiting -> existing

File: dev_guide/pod_autoscaling.adoc [hunk 3]
```diff
@@ -129,14 +126,13 @@ spec:
...
+<4> The API version of the object to scale.
+<5> The minimum number of replicas to which to scale down. The default is `1`.
```
Comment (bergerhoffer) [inline dev_guide/pod_autoscaling.adoc]: to which to scale down -> to scale down

File: dev_guide/pod_autoscaling.adoc [hunk 4]
```diff
@@ -129,14 +126,13 @@ spec:
...
+<6> The maximum number of replicas to which to scale up.
```
Comment (bergerhoffer) [inline dev_guide/pod_autoscaling.adoc]: to which to scale up -> to scale up

File: dev_guide/pod_autoscaling.adoc [hunk 5]
```diff
@@ -161,25 +157,34 @@ spec:
...
-<3> The kind of object to scale
-<4> The name of the object to scale
-<5> The minimum number of replicas to which to scale down
-<6> The maximum number of replicas to which to scale up
-<7> The average percentage of the requested CPU that each pod should be using
+<1> The name of this horizontal pod autoscaler object.
+<2> The API version of the object to scale.
+<3> The kind of object to scale.
+<4> The name of an existing object you want to scale.
+<5> The minimum number of replicas to which to scale down.
```
Comment (bergerhoffer) [inline dev_guide/pod_autoscaling.adoc]: to which to scale down -> to scale down

File: dev_guide/pod_autoscaling.adoc [hunk 6]
```diff
@@ -161,25 +157,34 @@ spec:
...
+<6> The maximum number of replicas to which to scale up.
```
Comment (bergerhoffer) [inline dev_guide/pod_autoscaling.adoc]: to which to scale up -> to scale up

File: dev_guide/pod_autoscaling.adoc [hunk 7]
```diff
@@ -161,25 +157,34 @@ spec:
...
+====
+
+To use memory-based autoscaling:
+
+. Enable memory-based autoscaling:
+
+.. Add the following to your
 cluster's `master-config.yaml` file:
 +
 [source,bash]
```
Comment (bergerhoffer) [inline dev_guide/pod_autoscaling.adoc]: I know this was already there, but should be source,yaml

File: dev_guide/pod_autoscaling.adoc [hunk 8]
```diff
@@ -191,6 +196,22 @@ apiServerArguments:
 ...
 ----
 
+.. Restart the {product-title} services:
++
+----
+# master-restart api
```
Comment (bergerhoffer) [inline dev_guide/pod_autoscaling.adoc]: This uses # as a prompt where other steps use $, is that correct, or should they all be the same?

File: dev_guide/pod_autoscaling.adoc [hunk 9]
```diff
@@ -202,23 +223,25 @@ metadata:
...
-<5> The minimum number of replicas to which to scale down
-<6> The maximum number of replicas to which to scale up
-<7> The average percentage of the requested memory that each pod should be using
+<1> The name of this horizontal pod autoscaler object.
+<2> The API version of the object to scale:
+* For a ReplicationController, use `v1`,
+* For a DeploymentConfig, use `apps.openshift.io/v1`.
+<3> The kind of object to scale, either `ReplicationController` or `DeploymentConfig`.
+<4> The name of an existing object you want to scale.
+<5> The minimum number of replicas to which to scale down.
```
Comment (bergerhoffer) [inline dev_guide/pod_autoscaling.adoc]: to which to scale down -> to scale down

File: dev_guide/pod_autoscaling.adoc [hunk 10]
```diff
@@ -202,23 +223,25 @@ metadata:
...
+<6> The maximum number of replicas to which to scale up.
```
Comment (bergerhoffer) [inline dev_guide/pod_autoscaling.adoc]: to which to scale up -> to scale up

**Review by bergerhoffer — COMMENTED** [review]
Just a few comments.

File: dev_guide/pod_autoscaling.adoc (hunk 3 above)
Comment (mburke5678) [reply dev_guide/pod_autoscaling.adoc]: I found better wording in the 4.x docs:
_The minimum number of replicas when scaling down._
_The maximum number of replicas when scaling up._

**Pull request merged**